src/cryptonote_basic/tx_extra.h

```diff
@@ -256,6 +257,48 @@ namespace cryptonote
     END_SERIALIZE()
   };
 
+  struct tx_extra_service_node_deregister_
+  {
+    struct vote
+    {
+      vote() = default;
+      vote(crypto::signature const &signature, uint16_t validator_index): signature(signature), validator_index(validator_index) { }
+      uint8_t           version = 0;
+      uint16_t          validator_index;
+      crypto::signature signature;
+
+      BEGIN_SERIALIZE()
+        VARINT_FIELD(version)
+        VARINT_FIELD(validator_index)
+        FIELD       (signature)
+      END_SERIALIZE()
+    };
+
+    uint8_t           version = 0;
+    // TODO(loki): vote_version/vote_type maps to quorum_vote_t version and
+    // type, but getting that declaration into tx_extra is not nice without
+    // having to bend backwards since tx_extra is included in everything, so
+    // trying to get it to include another file is a herculean task
+
+    // We need the data so that we can rederive the signature for clients to
+    // check the validity
+    uint8_t           vote_version;
+    uint8_t           vote_type;
+
+    uint64_t          block_height;
+    uint16_t          service_node_index;
+    std::vector<vote> votes;
+
+    BEGIN_SERIALIZE()
+      VARINT_FIELD(version)
+      VARINT_FIELD(vote_version)
+      VARINT_FIELD(vote_type)
```

Comment:
All these fields seem superfluous, and add precious bytes into the chain.  You could easily collapse these three into just one single `vote_type` field:
- 0 means: vote_type checkpoint
- 1 means: vote_type dereg
- 2 means: vote_type decommission
etc.

If later on a a decommission needs more data, you just create a:
- 3 means vote_type decommission, v2